pylint/testutils/reporter_for_tests.py

```diff
@@ -40,37 +42,44 @@ def handle_message(self, msg: Message) -> None:
             # 2to3 writes os.linesep instead of using
             # the previously used line separators
             str_message = str_message.replace("\r\n", "\n")
-        self.messages.append(f"{sigle}:{line:>3}{obj}: {str_message}")
+        self.messages.append(f"{sigle}:{line:>3}{obj}: {str_message}")  # type: ignore # Probably refactor at some point
 
-    def finalize(self):
+    def finalize(self) -> str:
         self.messages.sort()
         for msg in self.messages:
             print(msg, file=self.out)
-        result = self.out.getvalue()
+        if isinstance(self.out, StringIO):
+            result = self.out.getvalue()
```

Comment:
Either this is always true, in which case it shouldn't be there in the first place or `result` is possibly unbound.
For 1: Add `out: StringIO` below the class docstring.